### PR TITLE
Fixed #1593 - Missing changes on changes feed (Mac <- Android) pull r…

### LINKED ISSUE #1593: Missing changes on changes feed (Mac <- Android) pull replication
- Using Genymotion Google Nexus 4 API16 emulator, CBL 1.4-41
- Using Mac Sierra CBL 1.4-40

Scenario:
- Pull replication running on mac from android db to mac db
- Start continuous replication
- PUT 5000 docs to Android
- Verify all 5000 docs are pulled to mac by monitoring `_all_docs`

Expected:
All docs make it

Actual
4999 docs make it. Missing doc is `test_ls_db2_4828`

Possible duplicate of https://github.com/couchbase/couchbase-lite-java-core/issues/1495

To reproduce (Intermittentent, happened in a couple runs)

```
#!/usr/bin/env bash

#!/bin/bash
while :
do
    pytest --pdb -s --liteserv-one-platform=macosx --liteserv-one-version=1.4-40 --liteserv-one-host=192.168.0.14 --liteserv-one-port=50000 --liteserv-one-storage-engine=SQLite --liteserv-two-platform=android --liteserv-two-version=1.4-41 --liteserv-two-host=192.168.59.101 --liteserv-two-port=51000 --liteserv-two-storage-engine=SQLite -k "test_peer_2_peer_sanity_pull" testsuites/listener/shared/client_client/
    sleep 1
done
```

This will loop and break into the python debugger when the test fails ^

Mac listener logs - https://gist.github.com/sethrosetter/232e66d390538dbe29242b4f92f3e59d
Android listener logs - https://gist.github.com/sethrosetter/efd2ad7846419953a6f45c613cfd6737

### PR DESCRIPTION
**Pull request by hideki**
…eplication

Same issue with https://github.com/couchbase/couchbase-lite-java-core/issues/1495
Referenced previous PR: https://github.com/couchbase/couchbase-lite-java-core/pull/1521 , but this PR caused another another issue with `/_changes` REST API.

Notes:
- In case of `/_changes?feed=longpoll` REST API, there is case document is inserted or updated between calling `changesSince()` method and starting to observe Database change event. It could causing unhandled documents. This usually is caused by calling POST/PUT /{db}/{docID} and calling POST/GET /{db}/_changes?feed=longpool concurrently.
- To avoid above scenario, calling changesSince() method in DatabaseListener changed() callback method to make sure if any document changes occurs between them.
- Fixed the case startTimeout() is not called at line 1737.
- Make sure to set correct `since` value by deleting the method `sendLongpollChanges(rev)` that uses 0 as since value.

### CODE REVIEW

**Comment by hideki** [comment]
@pasin 
As QA is waiting this, so I will merge this PR myself.
If you have a chance, please review this?
Thanks!

File: src/main/java/com/couchbase/lite/router/Router.java [hunk 1]
```diff
@@ -1700,6 +1699,19 @@ public void changed(Database.ChangeEvent event) {
...
 
+            // In race condition, new doc or update doc is fired before starting to observe the
+            // DatabaseChangeEvent, it allows to skip few document changes with /_changes REST API.
+            // Make sure all document changes are tread by /_changes REST API.
+            if (!filled) {
+                filled = true;
+                RevisionList changes = db.changesSince(changesSince, changesOptions,
+                        changesFilter, changesFilterParams);
+                if (changes.size() > 0) {
+                    sendLongpollChanges(changes, changesSince);
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/router/Router.java]: What if it's a continuous feed?
* Should it check if it's longpoll before calling sendLogpollChanges?
* Does it issue not apply to continuous feed?

**Comment by pasin** [comment]
Sorry, I have been busy catching up 2.0.

File: src/main/java/com/couchbase/lite/router/Router.java (hunk 1 above)
Comment (hideki) [inline src/main/java/com/couchbase/lite/router/Router.java]: > Should it check if it's longpoll before calling sendLogpollChanges?

I believe it does not require to check it. DatabaseChangeListener is used for only feed=longpoll or continuous.

> Does it issue not apply to continuous feed?

I haven't seen that any CBL client replicator sends feed=continuous.

**Pull request merged**